Assets/00_Content/Scripts/Interactables/Beers/BeerTap.cs

```diff
@@ -4,25 +4,47 @@
 using UnityEngine.UI;
 
 namespace Interactables.Beers {
-
 	public class BeerTap : Interactable {
 
 		[Header("Settings")]
 		[Range(1, 100f)]
 		[SerializeField] private float pourTimeMultiplier = 10f;
+		[SerializeField] private int maxBeerAmount = 5;
+		[Tooltip("When the amount of beer left in the barrel goes below this amount the fillbar shows continually.")]
+		[SerializeField] private int showFillThreshold = 3;
 
 		[Header("GameObjects")]
 		[SerializeField] private GameObject beerPrefab;
 		[SerializeField] private Image progressBarImage;
 		[SerializeField] private GameObject progressBar;
+		[SerializeField] private Image fillBarImage;
+		[SerializeField] private GameObject fillBar;
 		[SerializeField] private BeerData beerData;
 
 		private ItemSlot itemSlot;
 		private float pouringProgress = 0;
 		private bool isHolding = false;
+		private int beerAmount;
+		private float fillPortion;
+
+		public int MaxBeerAmount => maxBeerAmount;
+
+		public int BeerAmount {
+			get { return beerAmount; }
+			set { beerAmount = Mathf.Clamp(value, 0, MaxBeerAmount); }
+		}
 
 		private void Start() {
 			itemSlot = GetComponentInChildren<ItemSlot>();
+
+			beerAmount = MaxBeerAmount;
+			fillPortion = 1f / MaxBeerAmount;
+		}
+
+		private void Update() {
+			fillBarImage.fillAmount = fillPortion * beerAmount;
+			if (beerAmount < showFillThreshold || pouringProgress > 0) fillBar.SetActive(true);
+			else fillBar.SetActive(false);
```

Comment:
I put it here mainly out of simplicity, and because I wanted to have the fillBar be shown continually if the amount of beer is less than the threshold.

Reply:
Would that functionality not stay the same if you just did the check after pouring a beer? Another occasion would also be when it's refilled though

Reply:
Yeah, then the functionality should be the same, I can change it so we can remove the Update() function.

Reply:
Done! Committing it now.